src/renderer/src/components/TransactionActivityList/index.tsx

```diff
@@ -0,0 +1,178 @@
+  const { handleScroll, ref: scrollRef } = useInfiniteScroll<HTMLDivElement>(fetchNextPage)
+
+  const groupedDataByDates = useMemo<TGroupedDataByDate[]>(() => {
+    const groupedData: TGroupedDataByDate[] = []
+
+    data.forEach(item => {
+      const formattedDate = format(item.date, 'dd MMM yyyy')
+      const currentGroupedData = groupedData.find(({ date }) => date === formattedDate)
+
+      if (!currentGroupedData) groupedData.push({ date: formattedDate, items: [item] })
+      else currentGroupedData.items.push(item)
+    })
+
+    return groupedData
+  }, [data])
+
+  const virtualizer = useVirtualizer({
+    count: groupedDataByDates.length,
+    gap: heights.DATE_GAP,
+    getScrollElement: () => scrollRef.current,
+    estimateSize: index => {
+      const { items } = groupedDataByDates[index] // Get the transactions (items) for the current date group
+      const itemsLength = items.length // Number of transactions (items) in this group
+
+      // Base height includes date label, separator, and separator margin
+      let height = heights.DATE + heights.SEPARATOR + heights.SEPARATOR_MARGIN
+
+      // If there aren't transactions (items), return the base height
+      if (itemsLength === 0) return height
+
+      // Add height for each header
+      height += itemsLength * heights.HEADER
+
+      // Add gaps between transactions (items), except after the last one
+      height += (itemsLength - 1) * heights.TRANSACTION_GAP
+
+      // Calculate total number of events across all items in the group
+      const eventsLength = items.flatMap(({ events }) => events).length
+
+      // Add height for each event
+      height += eventsLength * heights.EVENT
+
+      return height // Return the final estimated height
+    },
+  })
+
+  useEffect(() => {
+    virtualizer._willUpdate()
+  }, [virtualizer])
+
+  return (
+    <div className="w-full flex flex-col gap-y-2 mt-2 min-h-0 text-sm">
+      <p className="w-full text-xs font-thin text-gray-100">
+        {t('dates', {
+          dateFrom: format(dateFrom, 'dd/MM/yyyy'),
+          dateTo: format(dateTo, 'dd/MM/yyyy'),
+        })}
+      </p>
+
+      {match({ isLoading, groupedDataByDates })
+        .with({ isLoading: true }, () => (
+          <div className="flex flex-col overflow-y-auto min-h-0 w-full gap-y-6">
+            {new Array(4).fill(null).map((_, index) => (
+              <div key={`skeleton-${index}`} className="flex flex-col gap-y-2">
+                <Skeleton className="h-[40px]" />
+
+                <div className="flex gap-x-2 h-[34px]">
+                  <Skeleton className="h-full max-w-20" />
+                  <Skeleton className="h-full" />
+                </div>
+
+                <div className="flex flex-col gap-y-2 ml-20 pl-2">
+                  <Skeleton className="h-[50px]" />
+                  <Skeleton className="h-[50px]" />
+                  <Skeleton className="h-[50px]" />
+                </div>
+              </div>
```

Comment:
Create a new component called `TransactionActivityListSkeleton` and repeat here 4 times
```suggestion
              <TransactionActivityListSkeleton />
              <TransactionActivityListSkeleton />
              <TransactionActivityListSkeleton />
              <TransactionActivityListSkeleton />
```